src/quantuminspire/api.py

```diff
@@ -410,31 +619,69 @@ def execute_qasm(self, qasm, backend_type=None, number_of_shots=256, collect_tri
 
             has_results = self._wait_for_completed_job(quantum_inspire_job, collect_tries)
             return quantum_inspire_job.retrieve_results() if has_results else OrderedDict()
-
         finally:
-            if delete_project_afterwards:
+            if delete_project_afterwards and quantum_inspire_job is not None:
                 project_identifier = quantum_inspire_job.get_project_identifier()
                 self.delete_project(project_identifier)
 
-    def execute_qasm_async(self, qasm, backend_type=None, number_of_shots=256, default_number_of_shots=256,
-                           identifier=None, full_state_projection=True, project=None, job_name=None, user_data=''):
-        """ Creates the project, asset and job with the given qasm code and returns directly without waiting
-            for the job to complete.
+    def execute_qasm_async(self, qasm: str, backend_type: Union[OrderedDict, int, str] = None,
+                           number_of_shots:int = 256, default_number_of_shots:int = 256,
+                           identifier: str = None, full_state_projection: bool = True,
+                           project: OrderedDict = None, job_name: str = None, user_data: str = '') -> QuantumInspireJob:
+        """ With this method a cQASM program (job) is scheduled to be executed asynchronously. The method returns
+            directly without waiting for the job to complete, as opposed to method 'execute_qasm' which waits for
```

Comment:
```suggestion
            directly without waiting for the job to complete, as opposed to method `execute_qasm` which waits for
```

Reply:
Is there a convention for this?

Reply:
In mark-down, back-tics are used to mark code and I always think of method names and such as code. It makes them stand out and most people will recognize it as something that refers to code.